tests/bindings/test_bindings_nb.py

```diff
@@ -0,0 +1,213 @@
+        """Set up module attributes for all tests."""
+        for module_name in self.nb_modules:
+            self.module_attributes[module_name] = get_module_attributes(module_name)
+
+        for module_name in self.pb_modules:
+            self.module_attributes[module_name] = get_module_attributes(module_name)
+
+    def _skip_if_module_not_importable(self, module_name):
+        """Skip test if module is not importable.
+
+        This helper method checks if a module is importable and skips the current test
+        if it's not. It returns the module attributes for use in the test.
+
+        Each test that calls this will independently be skipped if the module is not available.
+        """
+        module_attr = self.module_attributes[module_name]
+        if not module_attr["importable"]:
+            pytest.skip(f"Module {module_name} not available: {module_attr.get('error')}")
+        return module_attr
+
+    @pytest.mark.parametrize("module_name", nb_modules)
+    def test_module_importable(self, module_name: str):
+        """Test if module can be imported."""
+        module_attr = self._skip_if_module_not_importable(module_name)
+        assert module_attr["importable"]
+        assert len(module_attr["classes"]) > 0 or len(module_attr["functions"]) > 0
+
+    @pytest.mark.parametrize("module_name", nb_modules)
+    def test_module_has_expected_info_dicts(self, module_name: str):
+        """Test if module has expected info dicts."""
+        module_attr = self._skip_if_module_not_importable(module_name)
+
+        # Check for backend_info if available
+        assert (
+            "compile_info" in module_attr["functions"]
+        ), f"compile_info not found in {module_name}"
+        assert (
+            "runtime_info" in module_attr["functions"]
+        ), f"runtime_info not found in {module_name}"
+
+    @pytest.mark.parametrize("module_name", nb_modules)
+    def test_statevector_classes_exists(self, module_name: str):
+        """Test if StateVectorC classes exists in the module."""
+        module_attr = self._skip_if_module_not_importable(module_name)
+
+        for precision in ["64", "128"]:
+            assert (
+                f"StateVectorC{precision}" in module_attr["classes"]
+            ), f"StateVectorC{precision} not found in {module_name}"
+
+    @pytest.mark.parametrize("module_name", nb_modules)
+    def test_exception_class_exists(self, module_name: str):
+        """Test if LightningException class exists in the module."""
+        module_attr = self._skip_if_module_not_importable(module_name)
+
+        assert (
+            "LightningException" in module_attr["classes"]
+        ), f"LightningException not found in {module_name}"
+
+    @pytest.mark.xfail(reason="Expected to fail while we don't have backend-specific bindings.")
+    @pytest.mark.parametrize("module_name, pybind_module_name", zip(nb_modules, pb_modules))
+    def test_api_parity_with_pybind(self, module_name, pybind_module_name):
```

Comment:
Is it possible to automatically check the function parameter types as well to ensure we don't accidentally have breaking API changes?

Reply:
Let me know if it looks good now.